spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Sigmoid.scala

```diff
@@ -43,8 +43,15 @@ class Sigmoid[@specialized(Float, Double) T: ClassTag](
   }
 
   override def updateGradInput(input: Tensor[T], gradOutput: Tensor[T]): Tensor[T] = {
-    gradInput.resizeAs(input).copy(gradOutput)
-    gradInput.cmul(output).cmul(buffer.add(ev.fromType(-1))).cmul(output)
+    updateGradInputInternal(output, gradOutput)
+  }
+
+  private[bigdl] def updateGradInputInternal(output: Tensor[T],
+                                             gradOutput: Tensor[T]): Tensor[T] = {
+    gradInput.resizeAs(gradOutput).copy(gradOutput)
```

Comment:
why change to this format?

Reply:
The tensorflow sigmoid grad use dy/dx = y(1-y) to calculate the gradient. The SigmoidGrad op will use the `updateGradInputInternal` function

Reply:
the result is equal. I wonder why to change to y(1-y) from original

Reply:
The original implementation needs a extra forward to fill in the `buffer`.